### PR TITLE
feat(NODE-3410): added MongoRuntimeError

### PR DESCRIPTION
**Pull request by andymina**
## Description
- Added `MongoRuntimeError` as per the proposed error class hierarchy

### CODE REVIEW

File: src/error.ts [hunk 1]
```diff
@@ -170,6 +170,25 @@ export class MongoDriverError extends MongoError {
   }
 }
 
+/**
+ * An error generated when the driver encounters unexpected unexpected input
+ * or reaches an unexpected/invalid internal state. Should never be
+ * directly instantiated.
```
Comment (nbbeeken) [inline src/error.ts:176]:
```suggestion
 * An error generated when the driver encounters unexpected unexpected input
 * or reaches an unexpected/invalid internal state. 
 * 
 * @privateRemarks
 * Should **never** be directly instantiated.
```

**Review by nbbeeken — CHANGES_REQUESTED** [review]
one small request, just no need to mention usage in public docs so you can put it under privateRemarks

**Review by nbbeeken — APPROVED** [review]
LGTM

File: src/error.ts [hunk 2]
```diff
@@ -170,6 +170,27 @@ export class MongoDriverError extends MongoError {
...
+ * or reaches an unexpected/invalid internal state
+ *
+ * @privateRemarks
+ * Should **never** be directly instantiated.
+ *
+ * @public
+ * @category Error
+ */
+export class MongoRuntimeError extends MongoDriverError {
+  code?: string;
```
Comment (dariakp) [inline src/error.ts:184]: MongoDriverError already has the code declared, so I don't think it's necessary to redeclare it on the subclass unless we're intending to remove it from the parent class; I also want to note that the code for MongoDriverError is used for exactly one error and I think there is a ticket that is meant to find a proper way to deal with that case (the ENOENT case); for now, I'd say just leave the explicit declaration off this class

**Review by dariakp — CHANGES_REQUESTED** [review]

File: src/error.ts (hunk 2 above)
Comment (dariakp) [reply src/error.ts:184]: (You could also consider making the actual codes a string enum for the more specific, directly instantiated, classes down the line)

**Review by W-A-James — APPROVED** [review]
LGTM

File: src/error.ts (hunk 2 above)
Comment (andymina) [reply src/error.ts]: Just removed the code from this class because it was redundant like you said.

File: src/error.ts [hunk 3]
```diff
@@ -170,6 +170,26 @@ export class MongoDriverError extends MongoError {
   }
 }
 
+/**
+ * An error generated when the driver encounters unexpected unexpected input
```
Comment (emadum) [inline src/error.ts:174]:
```suggestion
 * An error generated when the driver encounters unexpected input
```

**Review by emadum — CHANGES_REQUESTED** [review]
Caught a typo, LGTM other than that 👍

**Comment by andymina** [comment]
Changing this PR to be merged into NODE-3399 where Warren and I will implement the new error hierarchy

**Review by dariakp — APPROVED** [review]

**Review by emadum — APPROVED** [review]

**Pull request merged**